infoview/messages.tsx

```diff
@@ -0,0 +1,85 @@
+import { basename, escapeHtml, colorizeMessage } from './util';
+import { Message } from 'lean-client-js-node';
+import React = require('react');
+import { Location, DisplayMode, Config } from '../src/typings';
+import { MessagesContext, ConfigContext } from '.';
+import { DefaultSerializer } from 'v8';
+
+function compareMessages(m1: Message, m2: Message): boolean {
+    return (m1.file_name === m2.file_name &&
+        m1.pos_line === m2.pos_line && m1.pos_col === m2.pos_col &&
+        m1.severity === m2.severity && m1.caption === m2.caption && m1.text === m2.text);
+}
+
+export function MessageView(m: Message) {
+    // const f = escapeHtml(m.file_name);
+    const b = escapeHtml(basename(m.file_name));
+    const l = m.pos_line; const c = m.pos_col;
+    // const el = m.end_pos_line || l;
+    // const ec = m.end_pos_col || c;
+    // const cmd = encodeURI('command:_lean.revealPosition?' +
+        // JSON.stringify([Uri.file(m.file_name), m.pos_line, m.pos_col]));
+        // JSON.stringify([(m.file_name), m.pos_line, m.pos_col])); // [TODO] Uri.file isn't available in the webview?
+    const shouldColorize = m.severity === 'error';
+    let text = escapeHtml(m.text)
+    text = shouldColorize ? colorizeMessage(text) : text;
+    const title = `${b}:${l}:${c}`;
+    return <details open>
+        <summary className={m.severity + ' mv2 pointer'}>{title}</summary>
+        <div className="ml3">
+            <pre className="font-code" style={{whiteSpace: 'pre-wrap'}} dangerouslySetInnerHTML={{ __html: text }} />
```

Comment:
This doesn't enable line-wrapping for me, I still need to scroll horizontally:
```lean
example (h : 0 = by iterate 100 { refine _ * 1 }; exact 0) : true :=
by trivial
```

Somewhat more disturbingly, if I replace `trivial` by `trivia`, then I see *old* error messages from the line above ("unknown identifier 'exac'", etc.).

Reply:
Yes I'm also getting a bug where it is accumulating old error messages

Reply:
This is because I didn't set pre-wrap on the goal viewer in goal.tsx